packages/sdk/src/oracle/store/read.ts

```diff
@@ -159,8 +161,19 @@ export default class Read {
     assert(time, "Current time not available on chain: " + chainId);
     return time;
   };
+  sortedRequestsService = (): SortedRequests => {
+    const result = this.state?.services?.sortedRequests;
+    assert(result, "Sorted request service not set");
+    return result;
+  };
   oracleEvents = (chainId: number): OptimisticOracleEvent[] => {
     const chain = this.state?.chains?.[chainId];
     return chain?.optimisticOracle?.events || [];
   };
+  listChains = (): number[] => {
+    return Object.keys(this.state?.chains || {}).map(Number);
+  };
+  descendingRequests = (): RequestIndexes => {
+    return this.state.descendingRequests || [];
```

Comment:
This is the request list for the index page. frontend client can access this state directly or call the reader to get it.